web/models/survey.ex

```diff
@@ -413,22 +413,19 @@ defmodule Ask.Survey do
   end
 
   def stats(survey) do
-    respondents_by_state = survey |> respondents_by_state
-    respondents_total = Enum.map(respondents_by_state, fn {_, v} -> v end) |> Enum.reduce(fn q, acc -> q + acc end)
+    respondents_by_disposition = survey |> respondents_by_disposition
+    respondents_total = Enum.map(respondents_by_disposition, fn {_, v} -> v end) |> Enum.reduce(0, fn q, acc -> q + acc end)
     respondents_target = survey
       |> completed_respondents_needed_by
       |> respondents_target(respondents_total)
+    completion_rate = get_completion_rate(survey, respondents_by_disposition, respondents_target)
+    current_success_rate = get_success_rate(survey, respondents_by_disposition )
     initial_success_rate = initial_success_rate()
-    successful_respondents = survey |> successful_respondents(respondents_by_state)
-    current_success_rate = success_rate(successful_respondents, respondents_by_state["completed"], respondents_by_state["failed"], respondents_by_state["rejected"])
-    completion_rate = completion_rate(successful_respondents, respondents_target)
     estimated_success_rate = estimated_success_rate(initial_success_rate, current_success_rate, completion_rate)
-    completes = respondents_by_state["completed"]
-    pending = respondents_target - completes
+    completes = exhausted_respondents(respondents_by_disposition)
     multiplier = Float.ceil(1/estimated_success_rate, 0)
-    needed = pending*multiplier
-    selected_respondents = respondents_count(respondents_by_state)
-    missing = max(needed - selected_respondents, 0)
+    pending = not_exhausted_respondents(respondents_by_disposition)
+    {needed, missing} = get_needed_missing(survey, respondents_by_disposition, respondents_target, pending, completion_rate, current_success_rate)
```

Comment:
To calculate the needed and missing respondents, we have to use the _estimated_ success rate instead of the actual. We're estimating how many respondents are we missing, so it makes sense to use the estimated rate for that.